internal/database/recent_contribution_signal.go

```diff
@@ -189,6 +196,18 @@ const insertRecentContributorSignalFmtstr = `
 	) VALUES (%s, %s, %s, %s)
 `
 
+func (s *recentContributionSignalStore) ClearSignals(ctx context.Context, repoID api.RepoID) error {
+	tables := []string{"own_signal_recent_contribution", "own_aggregate_recent_contribution"}
+	q := "WITH rps as (select id from repo_paths where repo_id = %s) delete from %s where changed_file_path_id in (select * from rps);"
+
+	for _, table := range tables {
+		if err := s.Exec(ctx, sqlf.Sprintf(q, repoID, sqlf.Sprintf(table))); err != nil {
```

Comment:
Extract as a const + SQL casing

```suggestion
const clearSignalsFmtstr = `
    WITH rps AS (
        SELECT id FROM repo_paths WHERE repo_id = %s
    ) 
    DELETE FROM %s 
    WHERE changed_file_path_id IN (SELECT * FROM rps)
`

func (s *recentContributionSignalStore) ClearSignals(ctx context.Context, repoID api.RepoID) error {
	tables := []string{"own_signal_recent_contribution", "own_aggregate_recent_contribution"}

	for _, table := range tables {
		if err := s.Exec(ctx, sqlf.Sprintf(clearSignalsFmtstr, repoID, sqlf.Sprintf(table))); err != nil {
```